app/controllers/invites_controller.rb

```diff
@@ -0,0 +1,27 @@
+class InvitesController < ApplicationController
+  before_action :require_user
+
+  def new
+    @invitation = Invite.new
+  end
+
+  def create
+    @invitation = current_user.invites.new(invite_params)
+
+    if @invitation.save
+      @invitation.update_column(:invite_token, SecureRandom.urlsafe_base64)
+      UserMailer.send_invite(@invitation).deliver
+      flash[:success] = "Invitation has been sent!"
+      redirect_to new_invite_path
+    else
+      flash[:danger] = "Please fill in all the fields!"
```

Comment:
When you are about to `render` always use `flash.now` because it will appear on next request too. With `flash.now` it will appear right in the `render`.